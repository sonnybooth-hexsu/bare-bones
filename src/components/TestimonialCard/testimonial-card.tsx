import React from "react"

export const TestimonialCard = ({ testimonial }) => {
  return (
    <div className="bg-white pt-16 pr-16 pb-10 pl-24 shadow-md h-full">
      <div className="flex">
        <div className="rounded-full h-12 w-12 mr-2 mb-4 overflow-hidden">
          <img
            className="h-full w-full"
            src="https://i.pravatar.cc/150?img=32"
          />
        </div>
        <div>
          <h3 className="text-lg mb-1">Amelia</h3>
          <p className="text-xs leading-5 text-blue-400 font-semibold">
            Client
          </p>
        </div>
      </div>
      <p>{testimonial}</p>
    </div>
  )
}

export default TestimonialCard
