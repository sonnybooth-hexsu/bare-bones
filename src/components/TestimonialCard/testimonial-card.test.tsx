/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import { getByText, getByTestId } from "@testing-library/dom"
import { TestimonialCard } from "./testimonial-card"

afterEach(() => {
  cleanup()
})

describe("<ProductCard />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <TestimonialCard
        name="Mr Test"
        message="I am a test review"
        image="test-img.jpg"
        classes="mb-4"
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should have the title text passed through", () => {
    const { container } = render(
      <TestimonialCard
        name="Mr Test"
        message="I am a test review"
        image="test-img.jpg"
        classes="mb-4"
      />
    )

    const imgEl = getByTestId(container, "testimonialCardMessage")

    expect(getByText(imgEl, "I am a test review"))
  })
})
