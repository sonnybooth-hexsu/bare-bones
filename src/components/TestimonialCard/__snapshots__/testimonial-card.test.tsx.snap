// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should match snapshot 1`] = `
<div>
  <div
    class="bg-white pt-16 pr-16 pl-16 pb-10 md:pl-24 shadow-md h-full mb-4"
  >
    <div
      class="flex"
    >
      <div
        class="rounded-full mr-2 mb-4 overflow-hidden h-16 w-16"
      >
        <img
          alt="Testimonial client face"
          class="h-full w-full"
          src="/test-img.jpg"
        />
      </div>
      <div>
        <h3
          class="text-lg mb-1"
        >
          Mr Test
        </h3>
        <p
          class="text-xs leading-5 font-semibold"
        >
          Client
        </p>
      </div>
    </div>
    <p
      data-testid="testimonialCardMessage"
    >
      I am a test review
    </p>
  </div>
</div>
`;
